src/application/components/ClientNotFoundModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { Button } from "./Button";
+import { ButtonGroup } from "./ButtonGroup";
+import { Disclosure } from "./Disclosure";
+import { GitHubIssueLink, SECTIONS } from "./GitHubIssueLink";
+import { Modal } from "./Modal";
+
+interface ClientNotFoundModalProps {
+  open: boolean;
+  onClose: () => void;
+  onRetry: () => void;
+}
+
+function ConnectToDevToolsOptionLink() {
+  return (
+    <a
+      rel="noreferrer noopener"
+      target="_blank"
+      href="https://www.apollographql.com/docs/react/api/core/ApolloClient#apolloclientoptions-connecttodevtools"
+    >
+      <code>connectToDevTools</code> option
+    </a>
+  );
+}
+
+export function ClientNotFoundModal({
+  open,
+  onClose,
+  onRetry,
+}: ClientNotFoundModalProps) {
+  return (
+    <Modal open={open} onClose={onClose} size="xl">
+      <Modal.Header>
+        <Modal.Title>Could not find client</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <p>
+          An Apollo Client instance was not found, either because an Apollo
+          Client instance was never created, or because it could not be
+          discovered. This is most commonly fixed by setting the{" "}
+          <ConnectToDevToolsOptionLink /> to <code>true</code>.
+        </p>
+        <p className="mt-4">
+          If this was reached in error, please try one of the following
+          suggestions or click the &quot;Retry&quot; button to try looking for
+          the client again.
+        </p>
+        <div className="mt-4 flex flex-col gap-2">
+          <Disclosure>
```

Comment:
Please review each of these for accuracy and helpfulness! If you have any suggestions I missed for this first round, let me know and I'd be happy to add them!

Reply:
These are great! What a big improvement to the troubleshooting UX 😄